w3ctrace/state.go

```diff
@@ -17,109 +18,116 @@ const maxKVPairs = 32
 // Length of entries that should be filtered first in case, if tracestate has more than `maxKVPairs` items
 const thresholdLen = 128
 
-// State is list of key=value pairs representing vendor-specific data in the trace context
-type State []string
+var instanaListMemberRegex = regexp.MustCompile("^\\s*" + VendorInstana + "\\s*=\\s*([^,]*)\\s*$")
 
-// ParseState parses the value of `tracestate` header. Empty list items are omitted.
-func ParseState(traceStateValue string) (State, error) {
-	states := filterEmptyItems(strings.Split(traceStateValue, ","))
-	if len(states) < maxKVPairs {
-		return states, nil
-	}
+// State is list of key=value pairs representing vendor-specific data in the trace context
+type State struct {
+	// The key-value pairs of other vendors. The "in" list member, if present, will be stored separately
+	// (Note: list member is the term the W3C trace context specfication uses for the key-value pairs.)
+	listMembers []string
 
-	itemsToFilter := len(states) - maxKVPairs
-	filteredStates := states[:0]
+	// The value for the Instana-specific list member ("in=...").
+	instanaTraceStateValue string
+}
 
-	i := 0
-	for ; itemsToFilter > 0 && i < len(states); i++ {
+// makeState creates a new State with the given values
+func MakeState(listMembers []string, instanaTraceStateValue string) State {
```

Comment:
A common convention is to use `NewState`.
If the function doesn't need to be used outside of this package, you can start with a lower case letter, so its visibility stays limited to it.

The same applies to any other function. Eventually we have to make them visible outside of their package if we want to, for instance, use them in the tests, which are a different package name.

Reply:
I'll rename it to NewState then 👍 

The function is not used in production code at all, but only in `state_test.go` and `context_test.go`. That's why it is currently public/upper case.

Any recommendations around naming or visibility for that scenario (a function that only needs to be visible in the tests)?

One alternative would be to make the two attributes of the struct exported, but my gut feeling is that this would be actually worse than exporting the NewState function.

Reply:
As far as I can see in the code base, it is ok to make it publicly visible in order to be used in the tests, as the package name is different.
Maybe @andriikushch can have better ideas for cases like these.